File: src/js/sdk/src/urlUtils.ts
export function isExternalCallAllowed(targetUrl: string, disableExternalCalls: boolean): boolean {
    if (!disableExternalCalls) {
        return true;
    }
    
    try {
        const parsed = new URL(targetUrl);
        const hostLower = parsed.hostname.toLowerCase();
        return !(hostLower === 'lukuid.com' || hostLower.endsWith('.lukuid.com'));
    } catch (e) {
        // If we can't parse it, we allow it to proceed and let the fetch failure handle it.
        return true;
    }
}

File: src/ios/LukuIDSDK/Sources/LukuIDSDK/UrlUtils.swift
import Foundation

internal func isExternalCallAllowed(targetUrl: String, disableExternalCalls: Bool) -> Bool {
    if !disableExternalCalls {
        return true
    }
    
    guard let url = URL(string: targetUrl), let host = url.host?.lowercased() else {
        // If we can't parse it, we allow it to proceed and let the fetch failure handle it.
        return true
    }
    
    return !(host == "lukuid.com" || host.hasSuffix(".lukuid.com"))
}
